packages/vue-components/lib/components/MapboxMap/MapboxMap.vue

```diff
@@ -43,30 +43,51 @@ export default {
     mapboxStyle: {
       type: String,
       default: MAPBOX_STYLE
-    }
+    },
+    padding: {
+      type: Object,
+      default: () => ({ left: 0, right: 0, top: 0, bottom: 0 }),
+      validator: value => {
+        return Object.prototype.hasOwnProperty.call(value, 'top') || 
+               Object.prototype.hasOwnProperty.call(value, 'right') || 
+               Object.prototype.hasOwnProperty.call(value, 'bottom') || 
+               Object.prototype.hasOwnProperty.call(value, 'left')
+      }
+    },
+  },
+  watch: {
+    padding: {
+      handler: function(newValue) {
+        this.$refs.vmapbox.map.setPadding(newValue)
+      },
+      deep: true,
+    },
   },
   mounted() {
+    const map = this.$refs.vmapbox.map
```

Comment:
Would it be an idea to identify this constant as a DOM element? I would use `$map` as a fairly known convention, but some people don't like this, or argue against it because it's also used for vue instance properties. But then `mapEl` or something other wildly creative name could be thought of.

Reply:
I don't quite get why we would use a DOM element. I specifically interact with the mapbox instance, not the DOM element right?

Reply:
ups sorry, my bad